src/components/PostDetails.tsx

```diff
@@ -2,91 +2,47 @@
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-import * as commentsApi from '../api/comments';
-
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import {} from 'react-redux';
+import { RootState } from '../app/store';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addComment,
+  fetchCommentsByPost,
+  removeComment,
+} from '../features/comments';
 
 type Props = {
   post: Post;
 };
 
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+export const PostDetails: React.FC<Props> = () => {
+  const dispatch = useAppDispatch();
+  const { post } = useAppSelector(state => state.selectedPost);
```

Comment:
Consider specifying the `RootState` type in the `useAppSelector` hook to ensure type safety and avoid potential type inference issues.